src/test/java/org/broadinstitute/hellbender/utils/iterators/AllLocusIteratorUnitTest.java

```diff
@@ -0,0 +1,58 @@
+package org.broadinstitute.hellbender.utils.iterators;
+
+import org.broadinstitute.hellbender.engine.AlignmentContext;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+
+public class AllLocusIteratorUnitTest extends BaseTest {
+
+    @Test
+    public void testIteratorReturnsPileupsForAllLoci() {
```

Comment:
Add tests for intervals that lie before and after the interval

Reply:
Also tests for the sanity checks in the constructor (like having the wrong contig)

Reply:
Done!